scripts/o3de/ExportScripts/export_source_android.py

```diff
@@ -0,0 +1,383 @@
+                           seedlist_paths: List[pathlib.Path],
+                           seedfile_paths: List[pathlib.Path],
+                           level_names: List[str],
+                           asset_pack_mode: str = 'PAK',
+                           engine_centric: bool = False,
+                           should_build_tools: bool = True,
+                           should_build_all_assets: bool = True,
+                           build_config: str = 'profile',
+                           tool_config: str = 'profile',
+                           tools_build_path: pathlib.Path | None =None,
+                           max_bundle_size: int = 2048,
+                           fail_on_asset_errors: bool = False,
+                           deploy_to_device: bool = False,
+                           org_name: str|None = None,
+                           activity_name:str|None = None,
+                           logger: logging.Logger|None = None) -> None:
+    
+    expected_platform = 'android'
+    project_path = ctx.project_path
+    engine_path = ctx.engine_path
+    project_name = ctx.project_name
+
+    if not logger:
+        logger = logging.getLogger()
+        logger.setLevel(logging.ERROR)
+
+    is_installer_sdk = manifest.is_sdk_engine(engine_path=engine_path)
+
+    is_pc = exp.get_default_asset_platform() == 'pc'
+
+    o3de_cli_script_path = engine_path / ('scripts/o3de' + ('.bat' if is_pc else '.sh'))
+
+    assert exp.process_command([o3de_cli_script_path, 'android-configure', '--validate'], cwd=project_path) == 0, "Android config validation went wrong!"
+
+    # Calculate the tools and game build paths
+    default_base_path = engine_path if engine_centric else project_path
+
+    # Resolve (if possible) and validate any provided seedlist files
+    validated_seedslist_paths = exp.validate_project_artifact_paths(project_path=ctx.project_path,
+                                                                    artifact_paths=seedlist_paths)
+    
+    # Preprocess the seed file paths in case there are any wildcard patterns to use
+    preprocessed_seedfile_paths = exp.preprocess_seed_path_list(project_path=ctx.project_path,
+                                                                paths=seedfile_paths)
+    
+    # Convert level names into seed files that the asset bundler can utilize for packaging
+    for level in level_names:
+        preprocessed_seedfile_paths.append(ctx.project_path / f'Cache/{expected_platform}/levels' / level.lower() / (level.lower() + ".spawnable"))
+    
+    if is_installer_sdk:
+        # Check if we have any monolithic entries if building monolithic
+        if not exp.has_monolithic_artifacts(ctx):
+            logger.error("Trying to create monolithic build, but no monolithic artifacts are detected in the engine installation!.")
+            raise exp.ExportProjectError("Trying to build monolithic without libraries.")
```

Comment:
Well, this will fail for android. We don't have android monolithic libraries for the installer yet. And no need to log an error AND raise an exception. The exception can have the detailed error message that will be presented to the user instead.

Reply:
noted. I will remove that particular check for now, and we will revisit it once android monolithic libs are added for installers.